test/e2e/catalogsource_install_test.go

```diff
@@ -0,0 +1,323 @@
+package e2e
```

Comment:
An e2e without the real OLMv0 catalog operator might not be super useful. WDYT?

I wonder if we could simplify and still get most of the coverage by moving the tests to a set of catalogsource reconciler unit tests, where we could run a localhost grpc server directly in the unit test process (no container image, pod, or kubelet, or catalog operator necessary), create a catalog source pointing to our localhost grpc server, run Reconcile, and then assert on the expected ListEntities output. That would cover "this thing returns the entities we expect based on the state of a catalog source and grpc server.